solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -1430,9 +1430,11 @@ private void refreshAndWatchChildren() throws KeeperException, InterruptedExcept
       Stat stat = new Stat();
       List<String> replicaStates = null;
       try {
-        replicaStates = zkClient.getChildren(collectionPath, this, stat, true);
-        PerReplicaStates newStates = new PerReplicaStates(collectionPath, stat.getCversion(), replicaStates);
         DocCollection oldState = watchedCollectionStates.get(coll);
+        replicaStates = zkClient.getChildren(collectionPath, this, stat, true);
+        PerReplicaStates newStates = PerReplicaStates.fetch (collectionPath, zkClient, oldState == null ?
```

Comment:
I just realized that `PerReplicaStates.fetch` method makes calls to zk. And we are already fetching children in previous line here `replicaStates = zkClient.getChildren(collectionPath, this, stat, true);`

Reply:
true. good catch